sdks/go/test/integration/io/fhirio/fhirio_test.go

```diff
@@ -178,102 +185,91 @@ func extractResourcePathFrom(resourceLocationURL string) (string, error) {
 	return resourceLocationURL[startIdx:endIdx], nil
 }
 
-// Useful to prevent flaky results.
-func runWithBackoffRetries(t *testing.T, p *beam.Pipeline) error {
+func readTestTask(t *testing.T, s beam.Scope, testStoreInfo fhirStoreInfo) func() {
 	t.Helper()
 
-	var err error
-	for attempt := 0; attempt < len(backoffDuration); attempt++ {
-		err = ptest.Run(p)
-		if err == nil {
-			break
-		}
-		t.Logf("backoff %v after failure", backoffDuration[attempt])
-		time.Sleep(backoffDuration[attempt])
-	}
-	return err
-}
-
-func TestFhirIO_Read(t *testing.T) {
-	integration.CheckFilters(t)
-	checkFlags(t)
-
-	_, testResourcePaths, teardownFhirStore := setupFhirStoreWithData(t)
-	defer teardownFhirStore()
-
-	p, s, resourcePaths := ptest.CreateList(testResourcePaths)
-	resources, failedReads := fhirio.Read(s, resourcePaths)
-	passert.Empty(s, failedReads)
-	passert.Count(s, resources, "", len(testResourcePaths))
-
-	ptest.RunAndValidate(t, p)
-}
-
-func TestFhirIO_InvalidRead(t *testing.T) {
-	integration.CheckFilters(t)
-	checkFlags(t)
-
-	fhirStorePath, _, teardownFhirStore := setupFhirStoreWithData(t)
-	defer teardownFhirStore()
-
-	invalidResourcePath := fhirStorePath + "/fhir/Patient/invalid"
-	p, s, resourcePaths := ptest.CreateList([]string{invalidResourcePath})
-	resources, failedReads := fhirio.Read(s, resourcePaths)
+	testResources := append(testStoreInfo.resourcesPaths, testStoreInfo.path+"/fhir/Patient/invalid")
```

Comment:
Seeing as all these tests are now going to technically be part of the same pipeline, it may be helpful to call `s.Scope` on each of them to aid debugging.

Reply:
That's a good idea.